### ISSUE TITLE
4.0.1: I get the "The statement must be executed before any results can be obtained." message from the mssql 2012 server, when I call complex INSERT query.

### ISSUE THREAD

**Issue by ebadta81** [vert-x3/vertx-jdbc-client#209]
### Version

4.0.1

### Context

[My previous bug report](https://github.com/vert-x3/vertx-jdbc-client/issues/199) has been fixed, but now I get again this error message, when I try to execute a complex INSERT statement on an MSSQL server.

### Do you have a reproducer?

The sql statement that caused the error:

```sql
INSERT INTO testinsert (col1,col2,col3,createuser,updateuser)
SELECT ?,?,?,(select userid from login where username = ?),(select userid from login where username = ?)
FROM login l
JOIN login_role lr ON lr.login_userid = l.userid AND lr.role = 'admin'
WHERE l.username = ?
```

**Comment by pmlopes** [comment]
@ebadta81 could you help preparing a test for this?

**Comment by ebadta81** [comment]
Sure. What do you need? Complete sql with tables and data?

**Comment by pmlopes** [comment]
Yes, we already have a test containers setting up sqlserver, so that would speed up the troubleshooting: +1:

**Comment by ebadta81** [comment]
Okay, I tested with vertx-jdbc-client test container, and the error didn't occur.

Then I tested with my server, and the error came up.

I used the same tables, query, parameters, and changed the mssql driver version to 8.4.1.jre11 (this is what I use in my project).

The only difference is the mssql server version and configuration.

Unfortunately Microsoft do not offer mssql version 12 docker image...
